packages/app/obojobo-express/express_current_visit.js

```diff
@@ -8,6 +8,12 @@ const getCurrentVisitFromRequest = req => {
 		return Promise.resolve()
 	}
 
+	// In certain cases, events are sent via `navigator.sendBeacon`.
+	// Request's headers cannot be set. Therefore, `req.body` is still in JSON fotmat
+	try {
+		req.body = JSON.parse(req.body)
+	} catch (e) {} // eslint-disable-line no-empty
```

Comment:
This will run JSON.parse on any request getting the current visit, check if there's a way to limit it to when sendBeacon is used.